### PR TITLE
Login process stores user.

### PR DESCRIPTION
**Pull request by NihilistOwl**
Added cleanup to the beginnings of tests that use login to prevent errors.

Co-authored-by: Mitch Hardee <mitchell.hardee@coveros.com>
Co-authored-by: Jenna Charlton <jenna.charlton@coveros.com>
Co-authored-by: Byron Katz <byron.katz@coveros.com>
Co-authored-by: Matt Taylor <matthew.taylor@coveros.com>

### CODE REVIEW

**Review by coveros-matt-taylor — CHANGES_REQUESTED** [review]
10 Minute break first

File: src/main/kotlin/coverosR3z/persistence/PureMemoryDatabase.kt [hunk 1]
```diff
@@ -48,7 +48,7 @@ class PureMemoryDatabase {
 
     fun addNewUser(userName: UserName, hash: Hash, salt: String) : Int {
         val newIndex = users.size + 1
-        users.add(User(newIndex, userName.value, hash, salt))
+        users.add(User(newIndex, userName.value, hash, salt, null))
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/persistence/PureMemoryDatabase.kt:51]: Is this right? null for the employee id?

File: src/main/kotlin/coverosR3z/timerecording/TimeRecordingUtilities.kt [hunk 2]
```diff
@@ -10,6 +11,11 @@ import coverosR3z.persistence.EmployeeIntegrityViolationException
 class TimeRecordingUtilities(private val persistence: ITimeEntryPersistence) {
 
     fun recordTime(entry: TimeEntryPreDatabase): RecordTimeResult {
+        val user = CurrentUser.get()
+        // compare logged in user with user whose time is being entred, if no match, return error//
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/timerecording/TimeRecordingUtilities.kt]: slight syntactical concern

File: src/main/kotlin/coverosR3z/timerecording/TimeRecordingUtilities.kt [hunk 3]
```diff
@@ -10,6 +11,11 @@ import coverosR3z.persistence.EmployeeIntegrityViolationException
 class TimeRecordingUtilities(private val persistence: ITimeEntryPersistence) {
 
     fun recordTime(entry: TimeEntryPreDatabase): RecordTimeResult {
+        val user = CurrentUser.get()
```
Comment (7ep) [inline src/main/kotlin/coverosR3z/timerecording/TimeRecordingUtilities.kt]: could also add an invariant here - by golly, we better find a current user here.

**Review by coveros-matt-taylor — APPROVED** [review]

File: src/main/kotlin/coverosR3z/persistence/PureMemoryDatabase.kt (hunk 1 above)
Comment (NihilistOwl) [reply src/main/kotlin/coverosR3z/persistence/PureMemoryDatabase.kt:51]: At the moment, it seems that EmployeeID is not implemented. We will keep this in mind for proper implementation going forward, but will leave it in at this time.

**Review by coveros-matt-taylor — CHANGES_REQUESTED** [review]
Fix everything

**Review by 7ep — APPROVED** [review]
Looks entirely plausible.  This works.

**Review by coveros-matt-taylor — APPROVED** [review]

**Pull request merged**